### PR TITLE
Update map inits to account for Facing -> WAngle

### PR DESCRIPTION
**Pull request by pchote**
With all the plumbing changes out of the way, this is the first of two PRs to wrap up the facing rework by converting mod/map yaml to WAngle.

I am filing this PR now for visibility, but it currently depends on #18271 and #18264. Only the last 3 commits are new here.
I will hold off running the update rules until closer to merging to reduce conflicts if we merged any of @Smittytron's missions before then.

### CODE REVIEW

**Comment by pchote** [comment]
Rebased. I'd appreciate if we could get at least one :+1: on the code before I go through the pain of running the updater and then working out which oramap files need to be kept or unstaged.

File: OpenRA.Mods.Common/Scripting/Global/ActorGlobal.cs [hunk 1]
```diff
@@ -75,18 +75,17 @@ ActorInit CreateInit(string initName, LuaValue value)
 				return init;
 			}
 
-			// HACK: Forward compatibility for future WAngle facings
+			// HACK: Backward compatibility for legacy WAngle facings
```
Comment (abcdefg30) [inline OpenRA.Mods.Common/Scripting/Global/ActorGlobal.cs]: `legacy int facings`?

File: OpenRA.Mods.Common/Traits/Turreted.cs [hunk 2]
```diff
@@ -30,37 +30,27 @@ public class TurretedInfo : PausableConditionalTraitInfo, Requires<BodyOrientati
...
-		[Desc("Facing to use for actor previews (map editor, color picker, etc)")]
-		public readonly int PreviewFacing = 96;
-
 		[Desc("Display order for the turret facing slider in the map editor")]
 		public readonly int EditorTurretFacingDisplayOrder = 4;
 
 		IEnumerable<ActorInit> IActorPreviewInitInfo.ActorPreviewInits(ActorInfo ai, ActorPreviewType type)
 		{
-			yield return new TurretFacingInit(this, PreviewFacing);
+			// TODO: Update rule needs to account for PreviewFacing removal!
```
Comment (abcdefg30) [inline OpenRA.Mods.Common/Traits/Turreted.cs]: Can't this be removed now?

File: OpenRA.Mods.Common/Traits/Turreted.cs [hunk 3]
```diff
@@ -30,37 +30,27 @@ public class TurretedInfo : PausableConditionalTraitInfo, Requires<BodyOrientati
...
+			yield return new TurretFacingInit(this, WAngle.FromFacing(InitialFacing));
 		}
 
 		IEnumerable<EditorActorOption> IEditorActorOptions.ActorOptions(ActorInfo ai, World world)
 		{
-			yield return new EditorActorSlider("Turret", EditorTurretFacingDisplayOrder, 0, 255, 8,
+			yield return new EditorActorSlider("Turret", EditorTurretFacingDisplayOrder, 0, 1024, 8,
```
Comment (abcdefg30) [inline OpenRA.Mods.Common/Traits/Turreted.cs]: The old value inclines me to ask if this doesn't want to be `1023`.

File: OpenRA.Mods.Common/Traits/ThrowsParticle.cs [hunk 4]
```diff
@@ -66,20 +67,28 @@ public ThrowsParticle(ActorInitializer init, ThrowsParticleInfo info)
...
 			angle = new WAngle(Game.CosmeticRandom.Next(info.MinThrowAngle.Angle, info.MaxThrowAngle.Angle));
 			length = (finalPos - initialPos).Length / info.Velocity;
 
 			// Facing rotation
-			rotation = WAngle.FromFacing(WDist.FromPDF(Game.CosmeticRandom, 2).Length * info.TurnSpeed / 1024);
+			var rotationSpeed = WDist.FromPDF(Game.CosmeticRandom, 2).Length * info.TurnSpeed / 1024;
+			direction = 1;
+			if (rotationSpeed < 0)
+			{
+				rotation = -rotation;
```
Comment (abcdefg30) [inline OpenRA.Mods.Common/Traits/ThrowsParticle.cs]: I don't get this here: Why do you invert `rotation` here if it is getting reassign below regardless? (Mabye wants to be `rotationSpeed`?)

**Comment by pchote** [comment]
Updated. Fixed @abcdefg30's nits and updated TD and D2k maps. TS map updates will be handled in #18310 to avoid a rebase nightmare.

Holding off updating the RA maps until https://github.com/OpenRA/OpenRA/pulls/Smittytron have either been merged or explicitly marked as depending on this PR.

**Comment by obrakmann** [comment]
Smitty's RA maps have now all been merged, removing the dependencies label.

**Comment by pchote** [comment]
Updated. This should now be ready to go.

**Comment by abcdefg30** [comment]
But needs a rebase.

File: OpenRA.Mods.Common/Traits/Mobile.cs [hunk 5]
```diff
@@ -130,38 +130,13 @@ public bool CanStayInCell(World world, CPos cell)
 
 		IEnumerable<EditorActorOption> IEditorActorOptions.ActorOptions(ActorInfo ai, World world)
 		{
-			yield return new EditorActorSlider("Facing", EditorFacingDisplayOrder, 0, 255, 8,
+			yield return new EditorActorSlider("Facing", EditorFacingDisplayOrder, 0, 1024, 8,
```
Comment (abcdefg30) [inline OpenRA.Mods.Common/Traits/Mobile.cs]: Different place, same problem: `1023`

File: OpenRA.Mods.Common/Traits/ThrowsParticle.cs [hunk 6]
```diff
@@ -66,20 +67,29 @@ public ThrowsParticle(ActorInitializer init, ThrowsParticleInfo info)
...
 
 			// Facing rotation
-			rotation = WAngle.FromFacing(WDist.FromPDF(Game.CosmeticRandom, 2).Length * info.TurnSpeed / 1024);
+			var rotationSpeed = WDist.FromPDF(Game.CosmeticRandom, 2).Length * info.TurnSpeed / 1024;
+			direction = rotationSpeed < 0 ? -1 : 1;
+			if (rotationSpeed < 0)
+			{
+				// WAngle requires positive inputs, so track the direction separately
+				rotationSpeed = -rotationSpeed;
+				direction = -1;
```
Comment (abcdefg30) [inline OpenRA.Mods.Common/Traits/ThrowsParticle.cs]: `direction` is already set to `-1` at this point.

**Comment by pchote** [comment]
Fixed and rebased.

**Review by teinarss — APPROVED** [review]

**Review by abcdefg30 — APPROVED** [review]

**Comment by abcdefg30** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/58286871ee8df5ff7b7156b2cd4f4d4f92235b5b...f29a67c6d647cb3ebba797af664349e57b53816b)

**Pull request merged**